bevy_lint/docs/release.md

```diff
@@ -0,0 +1,51 @@
+# Release Checklist
+
+## Kick-off Pull Request
+
+1. Review the [changelog](../CHANGELOG.md) and ensure that all notable changes have been documented.
+2. Replace `[Unreleased]` heading with the version with the format `[X.Y.Z] - YYYY-MM-DD`.
+3. Update the `**All Changes**` link to compare from `main` to the new tag `lint-vX.Y.Z`. (E.g. `lint-v0.1.0...main` to `lint-v0.1.0...lint-v0.2.0`.)
+4. Remove the `-dev` suffix from the version in [`Cargo.toml`](../Cargo.toml), and ensure [`Cargo.lock`](../../Cargo.lock) also updates.
+5. Commit all of these changes and open a pull request.
+6. Merge the PR once a core Bevy maintainer approves it with no outstanding issues from other contributors.
+      - This starts the release process, enacting a freeze on all other changes until the release has finished.
```

Comment:
Is this freeze automatic or is this something maintainers just need to be aware of?

Reply:
Just something maintainers need to be aware of! This process should only take an hour or two, so it's not something that should ever be a problem.